frame/dapps-staking/src/testing_utils.rs

```diff
@@ -575,17 +579,156 @@ pub(crate) fn assert_claim_staker(claimer: AccountId, contract_id: &MockSmartCon
     );
 }
 
+pub(crate) fn assert_claim_staker_with_delegate(claimer: AccountId, delegated_account: AccountId, contract_id: &MockSmartContract<AccountId>) {
```

Comment:
Good that you added such function!

Perhaps it would be better to keep it a single `asset_claim_staker` though?
Right now you've introduced the second public function for the _same_ thing.